web3/contract.py

```diff
@@ -721,6 +746,99 @@ def __call__(self, *args, **kwargs):
+    is a subclass of this class.
+    """
+
+    web3 = None
+    contract_abi = None
+    contract_bytecode = None
+    transaction = None
+    args = None
+    kwargs = None
+
+    def __init__(self, abi, bytecode, web3, args=None, kwargs=None):
+
+        self.contract_abi = abi
+        self.contract_bytecode = bytecode
+        self.web3 = web3
+        self.args = args
+        self.kwargs = kwargs
+
+    def transact(self, transaction=None):
+
+        if transaction is None:
+            transact_transaction = {}
+        else:
+            transact_transaction = dict(**transaction)
+
+        if 'data' in transact_transaction:
+            raise ValueError("Cannot set data in constructor transaction")
+
+        if self.web3.eth.defaultAccount is not empty:
+            transact_transaction.setdefault('from', self.web3.eth.defaultAccount)
+
+        return transact_with_contract_constructor(self.contract_abi,
+                                                  self.contract_bytecode,
+                                                  self.web3,
+                                                  transact_transaction,
+                                                  *self.args,
+                                                  **self.kwargs)
+
+    def estimateGas(self, transaction=None):
+
+        if transaction is None:
+            estimate_transaction = {}
+        else:
+            estimate_transaction = dict(**transaction)
+
+        if 'data' in estimate_transaction:
+            raise ValueError("Cannot set data in constructor transaction")
+
+        if 'to' in estimate_transaction:
+            raise ValueError("Cannot set to in constructor transaction")
+
+        if self.web3.eth.defaultAccount is not empty:
+            estimate_transaction.setdefault('from', self.web3.eth.defaultAccount)
+
+        return estimate_gas_for_constructor(self.contract_abi,
+                                            self.contract_bytecode,
+                                            self.web3,
+                                            estimate_transaction,
+                                            *self.args,
+                                            **self.kwargs)
+
+    def buildTransaction(self, transaction=None):
+        """
+        Build the transaction dictionary without sending
+        """
+        if transaction is None:
+            built_transaction = {}
+        else:
+            built_transaction = dict(**transaction)
+
+        built_transaction = build_transaction_for_constructor(self.contract_abi,
+                                                 self.contract_bytecode,
+                                                 self.web3,
+                                                 built_transaction,
+                                                 *self.args,
+                                                 **self.kwargs)
+
+        built_transaction['to'] = ''
```

Comment:
Here was my suggestion to fix that bug. https://github.com/ethereum/web3.py/issues/739